marconi-chain-index/src/Marconi/ChainIndex/CLI.hs

```diff
@@ -317,17 +318,28 @@ commonMaybeTargetAssetParser =
 parseAsset :: Text -> Opt.ReadM (C.PolicyId, Maybe C.AssetName)
 parseAsset arg = do
   let parseAssetName :: Text -> Opt.ReadM C.AssetName
-      parseAssetName = either (fail . show) pure . C.deserialiseFromRawBytesHex C.AsAssetName . Text.encodeUtf8
+      parseAssetName =
+        either (fail . C.displayError) pure . C.deserialiseFromRawBytesHex C.AsAssetName . Text.encodeUtf8
 
       parsePolicyId :: Text -> Opt.ReadM C.PolicyId
-      parsePolicyId = either (fail . show) pure . C.deserialiseFromRawBytesHex C.AsPolicyId . Text.encodeUtf8
+      parsePolicyId =
+        either (fail . displayError') pure . C.deserialiseFromRawBytesHex C.AsPolicyId . Text.encodeUtf8
+
+      displayError' =
```

Comment:
Add a comment on why we need this vs just using `displayError`. You can just say like in your previous comment that we don't want to mention `ScriptHash` in the error message.